datafusion/sqllogictest/test_files/join.slt

```diff
@@ -1209,3 +1209,20 @@ drop table t1;
 
 statement ok
 drop table t2;
+
+# Test SQLancer issue: https://github.com/apache/datafusion/issues/12337
+statement ok
+create table t1(v1 int) as values(100);
+
+## Query with Ambiguous column reference
+query error DataFusion error: Schema error: Ambiguous reference to unqualified field v1
```

Comment:
I think it should be an "Ambiguous reference to qualified field", but currently the SchemaError cannot describe it, so the current situation is acceptable to me.

Reply:
How is this error different from `DuplicatedQualifiedField`?
https://github.com/apache/datafusion/blob/main/datafusion/common/src/error.rs#L149-L152

Reply:
`DuplicatedQualifiedField` was used to check the schema during its creation. After PR 6091, we allow duplicate qualified fields in the schema, so this error is no longer used.